src/ray/gcs/client.cc

```diff
@@ -24,6 +24,12 @@ Status Attach(plasma::EventLoop &event_loop) {
   return Status::OK();
 }
 
+Status AsyncGcsClient::AttachToAsio(boost::asio::io_service &io_service) {
+  asio_client_.reset(
```

Comment:
Maybe leave a note that we should only be attached to one event loop at a time (asio vs the `plasma::EventLoop`)?